lib/net/ldap/connection.rb

```diff
@@ -14,7 +14,7 @@ def initialize(server)
     rescue SocketError
       raise Net::LDAP::Error, "No such address or other socket error."
     rescue Errno::ECONNREFUSED
-      raise Net::LDAP::Error, "Server #{server[:host]} refused connection on port #{server[:port]}."
+      raise Net::LDAP::ConnectionRefusedError, "Server #{server[:host]} refused connection on port #{server[:port]}."
```

Comment:
I came to think no need to wrap `Errno::ECONNREFUSED` in this situation. 
@mtodd @jch  How do you think?